lib/views/walks/walks_view.dart

```diff
@@ -246,6 +251,30 @@ class _WalksViewState extends State<WalksView> with WidgetsBindingObserver {
     ScaffoldMessenger.of(context).showSnackBar(snackBar);
   }
 
+  void _news() async {
+    if (newsLaunched == false) {
+      List<dynamic> futures = await Future.wait(
+          [PrefsProvider.prefs.getString(Prefs.news), retrieveNews()]);
+
+      Set oldNews = futures[0] != null ? jsonDecode(futures[0]).toSet() : {};
+      List<News> news = futures[1];
+
+      int initialPage =
+          news.indexWhere((News news) => !oldNews.contains(news.name));
+
+      if (mounted && initialPage >= 0) {
+        setState(() => newsLaunched = true);
+
+        if (news.isNotEmpty) {
+          Set<int> viewed = await showNews(context, news, initialPage);
+          oldNews.addAll(viewed.map((int index) => news[index].name));
```

Comment:
It's a bit complicated, wouldn't it be better to just replace the old news with the new received ones?

Reply:
I wanted to be sure the news was at least visible once for the user before not showing it again. (Because there is a possibility there is multiple different news at once, which is why I implemented showing the news with a carousel). The current system is following:

Api gives news that should be shown to users (every news from the api has a startdate and enddate. Api call gets only news which is to be shown, so between those two timestamps).

If no news from api => No dialog box
If news from api but they have all been seen by user => no dialog box
if news from api and there are some which have not yet been seen by user => put the 'not seen news' first in the carousel and add remaining 'already seen news' to the end of carousel.

The question is also: 'Should we only show news once to user, or should news be re-shown every 5 days for ex.)

Reply:
I don't see any logic that checks if the news is in the time range? For what I see and understand, we will display the survey news forever, since the endDate is never read.

Reply:
The API handles the check for dates